### PR TITLE
Use CPP in full code base

### PR DESCRIPTION
**Pull request by andreybolonin**

### CODE REVIEW

File: src/RuntimeLoader/ContainerRuntimeLoader.php [hunk 1]
```diff
@@ -23,11 +23,8 @@
  */
 class ContainerRuntimeLoader implements RuntimeLoaderInterface
 {
-    private $container;
-
-    public function __construct(ContainerInterface $container)
+    public function __construct(private ContainerInterface $container)
```
Comment (fabpot) [inline src/RuntimeLoader/ContainerRuntimeLoader.php]:
```suggestion
    public function __construct(
        private ContainerInterface $container,
    )
```

**Review by fabpot — COMMENTED** [review]
Would you like to update the full code base?

**Comment by andreybolonin** [comment]
@fabpot ok, no problem )

**Comment by smnandre** [comment]
@fabpot regarding the fabbot false positive on deprecation messages, should we add a check on "symfony/symfoy" on the "DeprecationMessageChecker" ?  Or is this something that can be configured ?

File: src/Extension/SandboxExtension.php [hunk 2]
```diff
@@ -55,12 +52,7 @@ public function disableSandbox(): void
 
     public function isSandboxed(?Source $source = null): bool
     {
-        return $this->sandboxedGlobally || $this->sandboxed || $this->isSourceSandboxed($source);
-    }
-
-    public function isSandboxedGlobally(): bool
```
Comment (fabpot) [inline src/Extension/SandboxExtension.php]: Must be reverted

File: src/Extension/SandboxExtension.php [hunk 3]
```diff
@@ -21,16 +21,13 @@
...
-    private $policy;
-    private $sourcePolicy;
 
-    public function __construct(SecurityPolicyInterface $policy, $sandboxed = false, ?SourcePolicyInterface $sourcePolicy = null)
+    public function __construct(
+        private SecurityPolicyInterface $policy,
+        private bool $sandboxed = false,
+        private ?SourcePolicyInterface $sourcePolicy = null,
+    )
     {
```
Comment (fabpot) [inline src/Extension/SandboxExtension.php]: CS fix for all such changes:

```suggestion
    ) {
```

File: src/TemplateWrapper.php [hunk 4]
```diff
@@ -18,19 +18,17 @@
...
     /**
      * This method is for internal use only and should never be called
      * directly (use Twig\Environment::load() instead).
      *
      * @internal
      */
-    public function __construct(Environment $env, Template $template)
+    public function __construct(
+        private Environment $env,
+        private Template $template
```
Comment (fabpot) [inline src/TemplateWrapper.php]:
```suggestion
        private Template $template,
```

File: src/Token.php [hunk 5]
```diff
@@ -37,11 +33,12 @@ final class Token
     public const ARROW_TYPE = 12;
     public const SPREAD_TYPE = 13;
 
-    public function __construct(int $type, $value, int $lineno)
+    public function __construct(
+        private int $type,
+        private $value,
+        private int $lineno
```
Comment (fabpot) [inline src/Token.php]:
```suggestion
        private int $lineno,
```

File: src/TokenStream.php [hunk 6]
```diff
@@ -21,13 +21,13 @@
...
 final class TokenStream
 {
-    private $tokens;
     private $current = 0;
-    private $source;
 
-    public function __construct(array $tokens, ?Source $source = null)
+    public function __construct(
+        private array $tokens,
+        private ?Source $source = null
```
Comment (fabpot) [inline src/TokenStream.php]:
```suggestion
        private ?Source $source = null,
```

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Cache/FilesystemCache.php [hunk 7]
```diff
@@ -20,13 +20,12 @@ class FilesystemCache implements CacheInterface
 {
     public const FORCE_BYTECODE_INVALIDATION = 1;
 
-    private $directory;
-    private $options;
-
-    public function __construct(string $directory, int $options = 0)
+    public function __construct(
+        private string $directory,
```
Comment (stof) [inline src/Cache/FilesystemCache.php]: This should not use a promoted property as the value assigned in it is not the argument directly.

File: src/Markup.php [hunk 8]
```diff
@@ -18,13 +18,12 @@
  */
 class Markup implements \Countable, \JsonSerializable
 {
-    private $content;
-    private $charset;
-
-    public function __construct($content, $charset)
+    public function __construct(
+        private $content,
```
Comment (stof) [inline src/Markup.php]: Same here as the constructor does not assign the argument itself.

File: src/Markup.php [hunk 9]
```diff
@@ -18,13 +18,12 @@
...
+        private $charset,
```
Comment (stof) [inline src/Markup.php]: The charset must be `?string`

File: src/NodeVisitor/OptimizerNodeVisitor.php [hunk 10]
```diff
@@ -47,12 +47,13 @@ final class OptimizerNodeVisitor implements NodeVisitorInterface
...
     private $loops = [];
     private $loopsTargets = [];
-    private $optimizers;
 
     /**
      * @param int $optimizers The optimizer mode
      */
-    public function __construct(int $optimizers = -1)
+    public function __construct(
+        private int $optimizers = -1,
```
Comment (stof) [inline src/NodeVisitor/OptimizerNodeVisitor.php:55]: Why making it a promoted properties when the constructor still assigns it separately ? This makes no sense as it means the constructor assigns it twice.

File: src/Extension/SandboxExtension.php [hunk 11]
```diff
@@ -55,7 +51,7 @@ public function disableSandbox(): void
 
     public function isSandboxed(?Source $source = null): bool
     {
-        return $this->sandboxedGlobally || $this->sandboxed || $this->isSourceSandboxed($source);
```
Comment (fabpot) [inline src/Extension/SandboxExtension.php]: The two properties are still needed IIUC.

File: src/Extension/SandboxExtension.php [hunk 12]
```diff
@@ -21,16 +21,12 @@
...
-    private $sourcePolicy;
 
-    public function __construct(SecurityPolicyInterface $policy, $sandboxed = false, ?SourcePolicyInterface $sourcePolicy = null)
-    {
-        $this->policy = $policy;
-        $this->sandboxedGlobally = $sandboxed;
-        $this->sourcePolicy = $sourcePolicy;
+    public function __construct(
+        private SecurityPolicyInterface $policy,
+        private bool $sandboxed = false,
```
Comment (fabpot) [inline src/Extension/SandboxExtension.php]:
```suggestion
        private bool $sandboxedGlobally = false,
```
?

**Review by stof — CHANGES_REQUESTED** [review]

File: src/NodeVisitor/OptimizerNodeVisitor.php (hunk 10 above)
Comment (fabpot) [reply src/NodeVisitor/OptimizerNodeVisitor.php:55]: Line 69 (`$this->optimizers = $optimizers;`) should be removed

File: src/Cache/FilesystemCache.php (hunk 7 above)
Comment (fabpot) [reply src/Cache/FilesystemCache.php]: I agree, let's revert cases where we need to do the assignment in the constructor.

File: src/Extension/SandboxExtension.php [hunk 13]
```diff
@@ -21,16 +21,12 @@
 
 final class SandboxExtension extends AbstractExtension
 {
-    private $sandboxedGlobally;
-    private $sandboxed;
```
Comment (fabpot) [inline src/Extension/SandboxExtension.php]: This is still needed

**Review by fabpot — CHANGES_REQUESTED** [review]

**Comment by fabpot** [comment]
Thank you @andreybolonin.

**Pull request merged**